daemon.go

```diff
@@ -64,28 +63,24 @@ func (d *menderDaemon) shouldStop() bool {
 func (d *menderDaemon) Run() error {
 	// set the first state transition
 	var toState State = d.mender.GetCurrentState()
+	var currentState State
 	cancelled := false
 	for {
-		// If signal SIGUSR1 received, check for update straight away.
+		// If signal SIGUSR1 or SIGUSR2 is received, force the state-machine to the correct state.
 		select {
-		case sig := <-d.updateCheck:
-			log.Debugf("received signal: %t", sig)
-			_, err := LoadStateData(d.sctx.store)
-			// No previous state stored, means no update was in progress,
-			// and we can safely force an update check.
-			if err != nil && os.IsNotExist(err) {
-				log.Debugf("received signal: %t. Forcing device from initState to updateCheck\n", toState)
-				if d.mender.IsAuthorized() {
-					toState = updateCheckState
-				} else {
-					toState = initState
-				}
-				d.mender.SetNextState(toState)
+		case nState := <-d.forceToState:
+			switch toState.(type) {
+			case *IdleState, *CheckWaitState:
```

Comment:
As far as I understand, there will only be two states where we want to force the next state in the client.
1) From idle - The base case.
2) If we are in either a update-check- or inventory-update-wait state, then upon the state-machine getting woke, it will return checkWait state, which also should be forced, as in this case we do not care about the polling intervals. Alas, the next state should be forced.

All other states are left untouched.

Reply:
Agree, I like this method much better!